### PR TITLE
Color: little cleaning and optimization

### PR DESCRIPTION
**Pull request by Enzo-Demeulenaere**
little cleaning of Color

### CODE REVIEW

File: src/Colors/Color.class.st [hunk 1]
```diff
@@ -113,13 +113,13 @@ Color class >> colorFrom: parm [
 	"(Color colorFrom: #(0.0 0.0 1.0)) >>> (Color blue)"
 
 	| aColor firstParm |
-	(parm isKindOf: Color)
```
Comment (guillep) [inline src/Colors/Color.class.st:116]: All this smells like we need a double dispatch

File: src/Colors/Color.class.st [hunk 2]
```diff
@@ -622,7 +621,7 @@ Color class >> initializeIndexedColors [
 				[ grayVal := v / 32.0.
 				a
 					at: index
-					put: (Color
+					put: (self class
```
Comment (jecisc) [inline src/Colors/Color.class.st]:
```suggestion
					put: (self
```

We already are class side so this should be #self

File: src/Colors/Color.class.st [hunk 3]
```diff
@@ -113,13 +113,13 @@ Color class >> colorFrom: parm [
 	"(Color colorFrom: #(0.0 0.0 1.0)) >>> (Color blue)"
 
 	| aColor firstParm |
-	(parm isKindOf: Color)
+	(parm isKindOf: self class)
```
Comment (jecisc) [inline src/Colors/Color.class.st]:
```suggestion
	(parm isKindOf: self)
```

We are class side so this should be self

**Review by jecisc — CHANGES_REQUESTED** [review]

**Comment by jecisc** [comment]
Tests failures are not related

**Review by jecisc — APPROVED** [review]

**Pull request merged**